services/dynamic/dynamic-xml.tester.js

```diff
@@ -111,46 +133,18 @@ t.create('XML from url | invalid url')
     color: 'red',
   })
 
-t.create('XML from url | user color overrides default')
+t.create('request should set Accept header')
   .get(
-    '.json?url=https://services.addons.mozilla.org/en-US/firefox/api/1.5/addon/707078&query=/addon/name&color=10ADED'
+    `.json?${queryString.stringify({
+      url: exampleUrl,
+      query: "//book[@id='bk102']/title",
+    })}`
   )
-  .expectBadge({
-    label: 'custom badge',
-    message: 'IndieGala Helper',
-    color: '#10aded',
-  })
-
-t.create('XML from url | error color overrides default')
-  .get(
-    '.json?url=https://github.com/badges/shields/raw/master/notafile.xml&query=//version'
-  )
-  .expectBadge({
-    label: 'custom badge',
-    message: 'resource not found',
-    color: 'red',
-  })
-
-t.create('XML from url | error color overrides user specified')
-  .get('.json?query=//version&color=10ADED')
-  .expectBadge({
-    label: 'custom badge',
-    message: 'invalid query parameter: url',
-    color: 'red',
-  })
-
-let headers
-t.create('XML from url | request should set Accept header')
-  .get('.json?url=https://xml-test/api.xml&query=/name')
   .intercept(nock =>
-    nock('https://xml-test')
-      .get('/api.xml')
-      .reply(200, function(uri, requestBody) {
-        headers = this.req.headers
-        return '<?xml version="1.0" encoding="utf-8" ?><name>dynamic xml</name>'
-      })
+    nock('https://example.test', {
+      reqheaders: { accept: 'application/xml, text/xml' },
```

Comment:
This causes a "Mocks not net satisfied" assertion to be thrown if the Accept header doesn't match.